platform-sdk/docs/proposals/001-merkle-state-api/001-merkle-state-api.md

```diff
@@ -0,0 +1,74 @@
+# Merkle State API
+
+## Summary
+
+Refactor the Merkle tree so that it can be fully represented by the State API objects. Create a module that will provide simple and easy-to-use API for the Merkle tree.
+
+## Purpose and Context
+
+There are two issues that need to be addressed:
+
+* We need to have a public API for the Merkle tree that is simple and easy to use, residing in a separate module with a minimal set of dependencies.
+* The Merkle tree has to be fully expressed in terms of State API. Currently, `PlatformState` stands out and needs to be refactored.
+
+The goal of this design proposal is to ensure that one can interact with all the objects in the Merkle tree using the newly introduced State API.
```

Comment:
This statement does not match with the title of the proposal PR. I understand that the two concepts are interrelated, but it is not made clear in this document. Nowhere in the document are state proofs discussed. There are several purposes for this proposal:

1. Create an easy to use API for the entire state so that Block Nodes can interact with the state without having to understand the intricacies of merkle trees and merkleDB
2. Define the state in public format so that state proofs are publicly consumable
3. Reshape the tree into binary format to enable small state proofs

The title of the proposal indicates the purpose of all this is for state proofs. If that is true, all this should be described here.

Reply:
I removed this sentence.